dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -319,6 +321,84 @@ default String convertToJSONString(Object content) {
         return json;
     }
 
+    /**
+     * Takes in a list of strings ["image_1", "image_2",..., "image_n"] and returns a formatted string: "image_1, image_2, ... and image_n"
+     *
+     * @param images : A list of docker image names
+     * @return
+     */
+    default String formatImageStrings(final List<String> images) {
+        int lastIndex = images.size() - 1;
+        if (images.size() == 1) {
+            return images.get(lastIndex);
+        } else {
+            String imagesString = String.join(", ", images.subList(0, lastIndex));
+            imagesString = String.join(" and ", imagesString, images.get(lastIndex));
+            return imagesString;
+        }
+    }
+
+    /**
+     * Check that all images are specified using a digest or tag.
+     *
+     * @param versionName of the workflow that the snapshot is being requested for
+     * @param toolsJSONTable
+     * @throws CustomWebApplicationException if there is at least one image that is specified by 'latest' tag, no tag, or parameter.
+     */
+    default void checkSnapshotImages(final String versionName, final String toolsJSONTable) throws CustomWebApplicationException {
+        List<Map<String, String>> dockerTools = new ArrayList<>();
+        dockerTools = (ArrayList<Map<String, String>>)GSON.fromJson(toolsJSONTable, dockerTools.getClass());
```

Comment:
Real minor, unless you're running into compiler issues, I would combine line 349 and 350 -- you're creating an object on line 349 that immediately gets discarded on line 350.

It won't matter in terms of noticeable performance, but still, might as well get rid of it.